### PR TITLE
chore: bump elasticmq to v0.14.2

### PR DESCRIPTION
**Pull request by mberlanda**
I opened this morning an issue about upgrading elasticmq dependency: https://github.com/localstack/localstack/issues/879

It seems that after the merge of adamw/elasticmq#143 FIFO queues should be available so I tried to bump to a more recent version. Not sure I missed something but I tried to create one FIFO queue and it still seems to not work

*Step to reproduce*

```
$ docker build -t localstack:custom .
$ docker run -d -e SERVICES=sqs -p 4576:4576 localstack:custom
$ aws --endpoint-url=http://localhost:4576 sqs list-queues
# empty output without queues
```

I created then the following terraform configuration:

References:
https://github.com/localstack/localstack/issues/718
https://www.terraform.io/docs/providers/aws/r/sqs_queue.html

```tf
provider "aws" {
  region                      = "us-east-1"
  access_key                  = "anaccesskey"
  secret_key                  = "asecretkey"
  skip_credentials_validation = true
  skip_metadata_api_check     = true
  s3_force_path_style         = true

  endpoints {
    sqs      = "http://localhost:4576"
  }
}

resource "aws_sqs_queue" "terraform_queue" {
  name                        = "terraform-example-queue.fifo"
  fifo_queue                  = true
  content_based_deduplication = true
}
```
I saved this file in `tf/main.tf`

```
$ cd tf
$ terraform init -input=false
$ terraform plan -input=false
$ terraform apply -input=false -auto-approve

aws_sqs_queue.terraform_queue: Refreshing state... (ID: http://localhost:4576/queue/terraform-example-queue_fifo)
aws_sqs_queue.terraform_queue: Destroying... (ID: http://localhost:4576/queue/terraform-example-queue_fifo)
aws_sqs_queue.terraform_queue: Destruction complete after 0s
aws_sqs_queue.terraform_queue: Creating...
  arn:                               "" => "<computed>"
  content_based_deduplication:       "" => "true"
  delay_seconds:                     "" => "0"
  fifo_queue:                        "" => "true"
  kms_data_key_reuse_period_seconds: "" => "<computed>"
  max_message_size:                  "" => "262144"
  message_retention_seconds:         "" => "345600"
  name:                              "" => "terraform-example-queue.fifo"
  policy:                            "" => "<computed>"
  receive_wait_time_seconds:         "" => "0"
  visibility_timeout_seconds:        "" => "30"

Error: Error applying plan:

1 error(s) occurred:

* aws_sqs_queue.terraform_queue: 1 error(s) occurred:

* aws_sqs_queue.terraform_queue: [ERR] Error updating SQS attributes: InvalidAttributeName: InvalidAttributeName; see the SQS docs.
	status code: 400, request id: 00000000-0000-0000-0000-000000000000

Terraform does not automatically rollback in the face of errors.
Instead, your Terraform state file has been partially updated with
any resources that successfully completed. Please address the error
above and apply again to incrementally change your infrastructure.
```

Would I need any further step?

### CODE REVIEW

**Comment by mberlanda** [comment]
hey @simong

I found a reference to your work on FIFO queues support within elasticmq in this issue https://github.com/adamw/elasticmq/issues/125

Would you have any idea why the bump of version in localstack is not providing support for FIFO out of the box? Would you know if the example provided in the description would work against your code?

**Comment by mberlanda** [comment]
Oh my bad! I found it out:

https://github.com/localstack/localstack/blob/master/localstack/services/sqs/sqs_listener.py#L25-L27

I am going to fix it during the weekend!

File: localstack/services/sqs/sqs_listener.py [hunk 1]
```diff
@@ -21,13 +21,8 @@ def forward_request(self, method, path, data, headers):
         if method == 'POST' and path == '/':
             req_data = urlparse.parse_qs(to_str(data))
             if 'QueueName' in req_data:
-                if '.' in req_data['QueueName'][0]:
```
Comment (mberlanda) [inline localstack/services/sqs/sqs_listener.py]: Build failing on Travis. Linting unused import:
```
./localstack/services/sqs/sqs_listener.py:5:1: F401 'six.moves.urllib.parse.urlencode' imported but unused from six.moves.urllib.parse import urlencode ^ 
```

File: tests/integration/test_sqs.py [hunk 2]
```diff
@@ -16,3 +16,12 @@ def test_list_queue_tags(self):
 
         # Apparently, if there are no tags, then `Tags` should NOT appear in the response.
         assert 'Tags' not in res
+
+    def test_create_fifo_queue(self):
```
Comment (mberlanda) [inline tests/integration/test_sqs.py]: Build failing on Travis.

```
./tests/integration/test_sqs.py:21:5: W191 indentation contains tabs 	fifo_queue = 'my-queue.fifo' ^ 
./tests/integration/test_sqs.py:21:5: E101 indentation contains mixed spaces and tabs 	fifo_queue = 'my-queue.fifo' ^ 
./tests/integration/test_sqs.py:22:56: E999 TabError: inconsistent use of tabs and spaces in indentation sqs_client = aws_stack.connect_to_service('sqs') ^
./tests/integration/test_sqs.py:27:39: W292 no newline at end of file assert fifo_queue in queue_url ^
make[1]: *** [lint] Error 1 make[1]: Leaving directory `/home/travis/build/localstack/localstack' make: *** [test] Error 2
```

File: tests/integration/test_sqs.py [hunk 3]
```diff
@@ -16,3 +16,12 @@ def test_list_queue_tags(self):
...
+        fifo_queue = 'my-queue.fifo'
+        sqs_client = aws_stack.connect_to_service('sqs')
+        queue_info = sqs_client.create_queue(QueueName=fifo_queue)
```
Comment (mberlanda) [inline tests/integration/test_sqs.py]:
```
====================================================================
 ERROR: tests.integration.test_notifications.test_sqs_queue_names ---------------------------------------------------------------------- Traceback (most recent call last): File "/opt/code/localstack/.venv/lib/python2.7/site-packages/nose/case.py", line 197, in runTest self.test(*self.arg) File "/opt/code/localstack/tests/integration/test_notifications.py", line 30, in test_sqs_queue_names queue_url = sqs_client.create_queue(QueueName=queue_name)['QueueUrl'] File "/opt/code/localstack/.venv/lib/python2.7/site-packages/botocore/client.py", line 314, in _api_call return self._make_api_call(operation_name, kwargs) File "/opt/code/localstack/.venv/lib/python2.7/site-packages/botocore/client.py", line 612, in _make_api_call raise error_class(parsed_response, operation_name) ClientError: An error occurred (InvalidParameterValue) when calling the CreateQueue operation: InvalidParameterValue; see the SQS docs.
```

File: bin/Dockerfile.base [hunk 4]
```diff
@@ -26,16 +26,16 @@ RUN { \
     && chmod +x /usr/local/bin/docker-java-home
 ENV JAVA_HOME /usr/lib/jvm/java-1.8-openjdk
 ENV PATH $PATH:/usr/lib/jvm/java-1.8-openjdk/jre/bin:/usr/lib/jvm/java-1.8-openjdk/bin
-ENV JAVA_VERSION 8u131
-ENV JAVA_ALPINE_VERSION 8.131.11-r2
+ENV JAVA_VERSION 8u171
```
Comment (mberlanda) [inline bin/Dockerfile.base]: To fix:

```
Step 12/32 : RUN set -x && apk add --no-cache openjdk8="$JAVA_ALPINE_VERSION"     && [ "$JAVA_HOME" = "$(docker-java-home)" ]
+ apk add --no-cache 'openjdk8=8.131.11-r2'
fetch http://dl-cdn.alpinelinux.org/alpine/v3.8/main/x86_64/APKINDEX.tar.gz
fetch http://dl-cdn.alpinelinux.org/alpine/v3.8/community/x86_64/APKINDEX.tar.gz
ERROR: unsatisfiable constraints:
  openjdk8-8.171.11-r0:
    breaks: world[openjdk8=8.131.11-r2]
The command '/bin/sh -c set -x && apk add --no-cache openjdk8="$JAVA_ALPINE_VERSION"     && [ "$JAVA_HOME" = "$(docker-java-home)" ]' returned a non-zero code: 1
make: *** [docker-build-base] Error 1
```

File: bin/Dockerfile.base [hunk 5]
```diff
@@ -26,16 +26,16 @@ RUN { \
...
+ENV JAVA_ALPINE_VERSION 8.171.11-r0
 RUN set -x && apk add --no-cache openjdk8="$JAVA_ALPINE_VERSION" \
     && [ "$JAVA_HOME" = "$(docker-java-home)" ]
 
 # Install Maven - taken from official repo:
 # https://github.com/carlossg/docker-maven/blob/master/jdk-8/Dockerfile)
-ARG MAVEN_VERSION=3.5.2
+ARG MAVEN_VERSION=3.5.4
```
Comment (mberlanda) [inline bin/Dockerfile.base]: To fix:
```
Step 17/32 : RUN mkdir -p /usr/share/maven /usr/share/maven/ref   && curl -fsSL -o /tmp/apache-maven.tar.gz ${BASE_URL}/apache-maven-$MAVEN_VERSION-bin.tar.gz   && echo "${SHA}  /tmp/apache-maven.tar.gz" | sha256sum -c -   && tar -xzf /tmp/apache-maven.tar.gz -C /usr/share/maven --strip-components=1   && rm -f /tmp/apache-maven.tar.gz   && ln -s /usr/share/maven/bin/mvn /usr/bin/mvn
 ---> Running in 55c9c6f67b24
curl: (22) The requested URL returned error: 404 Not Found
The command '/bin/sh -c mkdir -p /usr/share/maven /usr/share/maven/ref   && curl -fsSL -o /tmp/apache-maven.tar.gz ${BASE_URL}/apache-maven-$MAVEN_VERSION-bin.tar.gz   && echo "${SHA}  /tmp/apache-maven.tar.gz" | sha256sum -c -   && tar -xzf /tmp/apache-maven.tar.gz -C /usr/share/maven --strip-components=1   && rm -f /tmp/apache-maven.tar.gz   && ln -s /usr/share/maven/bin/mvn /usr/bin/mvn' returned a non-zero code: 22
make: *** [docker-build-base] Error 22
```

References:

- https://apache.osuosl.org/maven/maven-3/ does not contain anymore 3.5.2
- https://maven.apache.org/download.cgi
- https://archive.apache.org/dist/maven/maven-3/3.5.4/binaries/

File: localstack/package.json [hunk 6]
```diff
@@ -4,7 +4,7 @@
   "description": "Local Cloud Stack and Utilities",
   "version": "0.0.1",
   "dependencies": {
-    "kinesalite": "1.13.0",
-    "leveldown": "1.6.0"
+    "kinesalite": "1.14.0",
```
Comment (mberlanda) [inline localstack/package.json]: Depends on `leveldown@1.9.0`

File: localstack/package.json [hunk 7]
```diff
@@ -4,7 +4,7 @@
...
+    "leveldown": "3.0.1"
```
Comment (mberlanda) [inline localstack/package.json]: To fix Nan node 10 issue:

```
Step 30/32 : RUN cd localstack && npm install
  CXX(target) Release/obj.target/leveldown/src/batch.o
In file included from ../../nan/nan.h:190:0,
                 from ../src/batch.cc:3:
../../nan/nan_maybe_43_inl.h: In function 'Nan::Maybe<bool> Nan::ForceSet(v8::Local<v8::Object>, v8::Local<v8::Value>, v8::Local<v8::Value>, v8::PropertyAttribute)':
../../nan/nan_maybe_43_inl.h:112:15: error: 'class v8::Object' has no member named 'ForceSet'
   return obj->ForceSet(isolate->GetCurrentContext(), key, value, attribs);
               ^~~~~~~~
In file included from ../src/batch.cc:3:0:
../../nan/nan.h: In function 'v8::Local<v8::Value> Nan::MakeCallback(v8::Local<v8::Object>, v8::Local<v8::Function>, int, v8::Local<v8::Value>*)':
../../nan/nan.h:833:60: warning: 'v8::Local<v8::Value> node::MakeCallback(v8::Isolate*, v8::Local<v8::Object>, v8::Local<v8::Function>, int, v8::Local<v8::Value>*)' is deprecated: Use MakeCallback(..., async_context) [-Wdeprecated-declarations]
         v8::Isolate::GetCurrent(), target, func, argc, argv);
                                                            ^
In file included from ../src/batch.cc:1:0:
/root/.node-gyp/10.8.0/include/node/node.h:172:50: note: declared here
                 NODE_EXTERN v8::Local<v8::Value> MakeCallback(
                                                  ^
/root/.node-gyp/10.8.0/include/node/node.h:88:42: note: in definition of macro 'NODE_DEPRECATED'
     __attribute__((deprecated(message))) declarator
                                          ^~~~~~~~~~
In file included from ../src/batch.cc:3:0:
../../nan/nan.h: In function 'v8::Local<v8::Value> Nan::MakeCallback(v8::Local<v8::Object>, v8::Local<v8::String>, int, v8::Local<v8::Value>*)':
../../nan/nan.h:848:62: warning: 'v8::Local<v8::Value> node::MakeCallback(v8::Isolate*, v8::Local<v8::Object>, v8::Local<v8::String>, int, v8::Local<v8::Value>*)' is deprecated: Use MakeCallback(..., async_context) [-Wdeprecated-declarations]
         v8::Isolate::GetCurrent(), target, symbol, argc, argv);
                                                              ^
In file included from ../src/batch.cc:1:0:
/root/.node-gyp/10.8.0/include/node/node.h:165:50: note: declared here
                 NODE_EXTERN v8::Local<v8::Value> MakeCallback(
                                                  ^
/root/.node-gyp/10.8.0/include/node/node.h:88:42: note: in definition of macro 'NODE_DEPRECATED'
     __attribute__((deprecated(message))) declarator
                                          ^~~~~~~~~~
In file included from ../src/batch.cc:3:0:
../../nan/nan.h: In function 'v8::Local<v8::Value> Nan::MakeCallback(v8::Local<v8::Object>, const char*, int, v8::Local<v8::Value>*)':
../../nan/nan.h:863:62: warning: 'v8::Local<v8::Value> node::MakeCallback(v8::Isolate*, v8::Local<v8::Object>, const char*, int, v8::Local<v8::Value>*)' is deprecated: Use MakeCallback(..., async_context) [-Wdeprecated-declarations]
         v8::Isolate::GetCurrent(), target, method, argc, argv);
                                                              ^
In file included from ../src/batch.cc:1:0:
/root/.node-gyp/10.8.0/include/node/node.h:158:50: note: declared here
                 NODE_EXTERN v8::Local<v8::Value> MakeCallback(
                                                  ^
/root/.node-gyp/10.8.0/include/node/node.h:88:42: note: in definition of macro 'NODE_DEPRECATED'
     __attribute__((deprecated(message))) declarator
                                          ^~~~~~~~~~
In file included from ../src/batch.cc:3:0:
../../nan/nan.h: In member function 'v8::Local<v8::Value> Nan::Callback::Call_(v8::Isolate*, v8::Local<v8::Object>, int, v8::Local<v8::Value>*) const':
../../nan/nan.h:1477:5: warning: 'v8::Local<v8::Value> node::MakeCallback(v8::Isolate*, v8::Local<v8::Object>, v8::Local<v8::Function>, int, v8::Local<v8::Value>*)' is deprecated: Use MakeCallback(..., async_context) [-Wdeprecated-declarations]
     ));
     ^
In file included from ../src/batch.cc:1:0:
/root/.node-gyp/10.8.0/include/node/node.h:172:50: note: declared here
                 NODE_EXTERN v8::Local<v8::Value> MakeCallback(
                                                  ^
/root/.node-gyp/10.8.0/include/node/node.h:88:42: note: in definition of macro 'NODE_DEPRECATED'
     __attribute__((deprecated(message))) declarator
                                          ^~~~~~~~~~
make: *** [leveldown.target.mk:119: Release/obj.target/leveldown/src/batch.o] Error 1
make: Leaving directory '/opt/code/localstack/localstack/node_modules/leveldown/build'
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:262:23)
gyp ERR! stack     at ChildProcess.emit (events.js:182:13)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:238:12)
gyp ERR! System Linux 4.9.87-linuxkit-aufs
gyp ERR! command "/usr/local/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /opt/code/localstack/localstack/node_modules/leveldown
gyp ERR! node -v v10.8.0
gyp ERR! node-gyp -v v3.7.0
gyp ERR! not ok 
```

**Comment by mberlanda** [comment]
@whummer great news! 

I managed to create FIFO queues from localstack with a custom branch in elasticmq:
https://github.com/adamw/elasticmq/pull/148

*Steps to reproduce*

- `sbt compile "project server" assembly` from the code in [this pr](https://github.com/adamw/elasticmq/pull/148) and rename the output (`/server/target/scala-2.12/elasticmq-server-assembly-0.14.3-SNAPSHOT.jar`) file to `elasticmq-server-custom.jar` or download the jar from
https://www.dropbox.com/s/h3aaxy5rg1jeopt/elasticmq-server-custom.jar?dl=0
- copy the jar into `bin` directory
- edit the Dockerfile.build as follows:

```
diff --git a/bin/Dockerfile.base b/bin/Dockerfile.base
index e83a52f..a41da85 100644
--- a/bin/Dockerfile.base
+++ b/bin/Dockerfile.base
@@ -56,14 +56,16 @@ ADD requirements.txt .
 RUN mkdir -p /opt/code/localstack/localstack/infra && \
     wget -O /tmp/localstack.es.zip \
         https://artifacts.elastic.co/downloads/elasticsearch/elasticsearch-6.2.0.zip && \
-    wget -O /tmp/elasticmq-server.jar \
-        https://s3-eu-west-1.amazonaws.com/softwaremill-public/elasticmq-server-0.14.2.jar && \
+    # wget -O /tmp/elasticmq-server.jar \
+    #     https://s3-eu-west-1.amazonaws.com/softwaremill-public/elasticmq-server-0.14.2.jar && \
     (cd localstack/infra/ && unzip -q /tmp/localstack.es.zip && \
         mv elasticsearch* elasticsearch && rm /tmp/localstack.es.zip) && \
     mkdir -p /opt/code/localstack/localstack/infra/dynamodb && \
     wget -O /tmp/localstack.ddb.zip \
         https://s3-us-west-2.amazonaws.com/dynamodb-local/dynamodb_local_latest.zip && \
     (cd localstack/infra/dynamodb && unzip -q /tmp/localstack.ddb.zip && rm /tmp/localstack.ddb.zip)
+
+ADD bin/elasticmq-server-custom.jar /tmp/elasticmq-server.jar
 RUN (pip install --upgrade pip) && \
     (test `which virtualenv` || pip install virtualenv || sudo pip install virtualenv) && \
     (virtualenv .testvenv && source .testvenv/bin/activate && \
```

```
$ make docker-build-base
$ docker build -t localstack:custom .
$ docker run -d -e SERVICES=sqs -p 4576:4576 localstack:custom
```

Then run terraform apply and it works.


Small question: Should I split this pr into two => upgrade dependencies and implement FIFO queue support? I see that the tests are not retrieving the custom `localstack/java-maven-node-python` image

**Comment by whummer** [comment]
I updated the base image and re-ran the builds on your PR - looks like all tests are passing now. Going to merge the changes now. Thanks again!

**Pull request merged**